### PR TITLE
Feature Domain Language Smells

### LINKED ISSUE #22: Expressiveness smells
Add smell detection for expressiveness: 
- used token should be full words in the target human language 
- used tokens should "long enough": more than, say, 3 chars long. 
- tokens should not be common meaningless terms, like tmp, x, foo, manager, and so on

Perhaps we should be able to deactive such smells detection sometimes.

### PR DESCRIPTION
**Pull request by flbulgarelli**
This PR incorporates into Mulang the ability of checking - at some level - expressiveness: 

* `HasTooShortBindings`: true if a binding is too short and not part of domain language's jargon
* `HasWrongCaseBindings`: true if a binding does not match the domain language's case style
* `HasMisspelledBindings`: true if a binding is not a domain language dictionary's word and not part of its jargon

All new smells are configurable, you can enable or disabled them, one by one.

Fixes #22

### CODE REVIEW

File: src/Language/Mulang/DomainLanguage.hs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                          dictionary :: Dictionary,
+                          caseStyle :: CaseStyle,
+                          minimumBindingSize :: Int,
+                          jargon :: [String] }
+
+type DomainLanguageInspection = DomainLanguage -> Inspection
+
+hasTooShortBindings :: DomainLanguageInspection
+hasTooShortBindings language = any isShort . mainDeclaredBindingsOf
+  where isShort binding = length binding < (minimumBindingSize language) && notJargonOf binding language
```
Comment (aguspina) [inline src/Language/Mulang/DomainLanguage.hs]: mmm that is not my definition of `short` 😝

File: spec/TokenizerSpec.hs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import           Test.Hspec
+import           Text.Inflections.Tokenizer (snakeCase, camelCase, tokenize)
+
+spec :: Spec
+spec = do
+  describe "tokenize" $ do
+    it "can tokenize camelCase words" $ do
+      tokenize camelCase "todayIsAGreatDay" `shouldBe` ["today", "is", "a", "great", "day"]
+
+    it "can tokenize snakeCase words" $ do
```
Comment (faloi) [inline spec/TokenizerSpec.hs]: Perhaps `snake_case`?

**Comment by flbulgarelli** [comment]
> I'm a bit sceptical about the accurateness of HasMisspelledBindings and HasTooShortBindings, obviously it would heavily depend on the dictionary. It would be nice to know how you're planning to build it (asuming that you have thought about it :stuck_out_tongue:).

As a first iteration about a Mumuki Platform integration, I am planning to just install both an Spanish and English dictionary in the same machines mulang runs. Then, as a second iteration, I am planning to build the `jargon` list either manually or from the description words not present in the dictionary. 

> Mulang does not specify a `default smells set`, but I am planning to build a specific default smells set per runner. That way we can search only for smells that are more common or more important in the target language.

File: spec/TokenizerSpec.hs (hunk 2 above)
Comment (flbulgarelli) [inline spec/TokenizerSpec.hs]: Ok. Fixing in my next PR

File: src/Language/Mulang/DomainLanguage.hs (hunk 1 above)
Comment (flbulgarelli) [inline src/Language/Mulang/DomainLanguage.hs]: Jargon words are never considered short

**Pull request merged**